src/sqlfluff/testing/rules.py

```diff
@@ -66,6 +67,14 @@ def assert_rule_fail_in_sql(code, sql, configs=None):
             "No {0} failures found in query which should fail.".format(code),
             pytrace=False,
         )
+    if line_numbers:
```

Comment:
Here, we add the ability to check the line number locations of the lint errors produced by a YAML test.